pkg/generate/app/cmd/newapp.go

```diff
@@ -710,20 +711,146 @@ func (c *AppConfig) Run() (*AppResult, error) {
 	}, nil
 }
 
+// followRefToDockerImage follows a buildconfig...To/From reference until it terminates in docker image information.
+// This can include dereferencing chains of ImageStreamTag references that already exist or which are being created.
+// ref is the reference to To/From to follow. If ref is an ImageStreamTag that is following another ImageStreamTag, isContext
+// should be set to the parent IS. Finally, objects is the list of objects that new-app is creating to support the buildconfig.
+// It returns a reference to a terminal DockerImage or nil if one could not be determined (a valid, non-error outcome). err
+// is only used to indicate that the follow encountered a severe error (e.g malformed data).
+func (c *AppConfig) followRefToDockerImage(ref *kapi.ObjectReference, isContext *imageapi.ImageStream, objects app.Objects) (*kapi.ObjectReference, error) {
+
+	if ref == nil {
+		return nil, fmt.Errorf("Unable to follow nil")
+	}
+
+	if ref.Kind == "DockerImage" {
+		// Make copy so we don't modify any structures new-app/build created
+		copy := *ref
```

Comment:
Aware this is not a deep copy, right? Should work as of today because `kapi.ObjectReference` is a struct with only string fields.